runtime/vm/vmprops.c

```diff
@@ -1040,13 +1040,22 @@ getJ9VMVersionString(J9JavaVM *vm) {
 UDATA
 addSystemProperty(J9JavaVM * vm, const char* propName,  const char* propValue, UDATA flags)
 {
+	char *convertedPropValue = NULL;
 	J9VMSystemProperty* newProp = pool_newElement(vm->systemProperties);
 	if (NULL == newProp) {
 		return J9SYSPROP_ERROR_OUT_OF_MEMORY;
 	}
 
+	/* Convert the value string from the platform encoding to the modified UTF8 */
+	convertedPropValue = (char *)getMUtf8String(vm, propValue, strlen(propValue));
+	if (NULL == convertedPropValue) {
+		PORT_ACCESS_FROM_JAVAVM(vm);
+		j9mem_free_memory(newProp);
```

Comment:
This is incorrect - should be `pool_removeElement`.

Reply:
The fix here will to be moved out as suggested by Pete.